projects/plugins/jetpack/extensions/shared/wait-for-editor.js

```diff
@@ -7,17 +7,25 @@ import { select, subscribe } from '@wordpress/data';
  */
 export const waitForEditor = async () =>
 	new Promise( resolve => {
-		const unsubscribe = subscribe( () => {
-			const isCleanNewPost = select( 'core/editor' ).isCleanNewPost();
+		// Resolve immediately if it's a clean new post or has blocks
+		if (
+			select( 'core/editor' ).isCleanNewPost() ||
+			select( 'core/block-editor' ).getBlocks().length > 0
+		) {
+			resolve();
+			return;
+		}
 
-			if ( isCleanNewPost ) {
-				unsubscribe();
-				resolve();
-			}
+		// Otherwise wait for blocks to appear
+		const timeoutId = setTimeout( () => {
+			unsubscribe();
+			resolve();
+		}, 2000 );
 
+		const unsubscribe = subscribe( () => {
 			const blocks = select( 'core/block-editor' ).getBlocks();
-
 			if ( blocks.length > 0 ) {
+				clearTimeout( timeoutId );
```

Comment:
Would it make sense to check isCleanNewPost here as well?

Reply:
I had to revert that change -- for some reason checking `isCleanNewPost` too makes it fail intermittently again. 😢